src/main/java/com/polishuchenko/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.polishuchenko.bookstore.service.shoppingcart.impl;
+
+import com.polishuchenko.bookstore.dto.cartitem.CartItemRequestDto;
+import com.polishuchenko.bookstore.dto.cartitem.CartItemResponseDto;
+import com.polishuchenko.bookstore.dto.cartitem.UpdateCartItemDto;
+import com.polishuchenko.bookstore.dto.shoppingcart.ShoppingCartDto;
+import com.polishuchenko.bookstore.exception.EntityNotFoundException;
+import com.polishuchenko.bookstore.mapper.CartItemMapper;
+import com.polishuchenko.bookstore.mapper.ShoppingCartMapper;
+import com.polishuchenko.bookstore.model.Book;
+import com.polishuchenko.bookstore.model.CartItem;
+import com.polishuchenko.bookstore.model.ShoppingCart;
+import com.polishuchenko.bookstore.model.User;
+import com.polishuchenko.bookstore.repository.book.BookRepository;
+import com.polishuchenko.bookstore.repository.cartitem.CartItemRepository;
+import com.polishuchenko.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import com.polishuchenko.bookstore.service.shoppingcart.ShoppingCartService;
+import com.polishuchenko.bookstore.service.user.UserService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserService userService;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public CartItemResponseDto update(Long id, UpdateCartItemDto updateCartItemDto) {
+        CartItem cartItem = cartItemRepository
+                .findCartItemByIdAndShoppingCartId(id, currentUserCart().getId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find item with id= " + id));
+        cartItem.setQuantity(updateCartItemDto.getQuantity());
+        cartItem.setId(id);
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public void delete(Long id) {
+        cartItemRepository.deleteById(id);
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCart() {
+        return shoppingCartMapper.toDto(currentUserCart());
+    }
+
+    @Transactional
+    @Override
+    public CartItemResponseDto addCartItem(CartItemRequestDto cartItemRequestDto) {
```

Comment:
To many responsibilities for this method, I suggest moving logic or increasing quantity of cartItem in separate method